WordPress/gutenberg
# DropdownMenu V2 tweaks

## What?


Closes #55933

This PR includes a series of tweaks to the experimental new version of `DropdownMenu`, based on the specs shared in #55933.

Main tweaks:
- Removed group label component (not needed)
- Added new subcomponents for menu item label and help text (mostly needed because of text truncation behavior)
- Added auto-indentation of menu items based on the presence, in the same popover, of at least one item with a prefix
- Tweaks to spacing, typography, colors, hovered/focused/disabled states...

## Why?


The design spec brings a coherent look with the rest of the components library, and addresses a few complex scenarios when dealing with some combinations of menu items and their contents.

## How?


The most complex bits in the implementation:
- CSS grid and subgrid were used to implement the auto-indent functionality
- The Label and Help text subcomponent use the `Truncate` component under the hood

## Testing Instructions





Open the "DropdownMenu v2 ariakit" Storybook examples, make sure that the component follows the design specs

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1083581/398c4224-83f2-4219-b14b-3d739630c75c
